sdk/src/main/java/com/mapbox/maps/ViewAnnotationManagerImpl.kt

```diff
@@ -0,0 +1,232 @@
+package com.mapbox.maps
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewTreeObserver
+import android.widget.FrameLayout
+import androidx.annotation.LayoutRes
+import androidx.annotation.VisibleForTesting
+import androidx.asynclayoutinflater.view.AsyncLayoutInflater
+import com.mapbox.bindgen.Expected
+import com.mapbox.maps.viewannotation.ViewAnnotation
+import com.mapbox.maps.viewannotation.ViewAnnotationManager
+import java.util.concurrent.ConcurrentHashMap
+import kotlin.collections.HashMap
+
+internal class ViewAnnotationManagerImpl(
+  private val mapView: MapView
+) : ViewAnnotationManager, ViewAnnotationPositionsUpdateListener {
+
+  private val mapboxMap: MapboxMap = mapView.getMapboxMap()
+  private val viewPlugins = mapView.mapController.pluginRegistry.viewPlugins
+
+  init {
+    mapView.requestDisallowInterceptTouchEvent(false)
+    mapboxMap.setViewAnnotationPositionsUpdateListener(this)
+  }
+
+  private val annotationMap = ConcurrentHashMap<String, ViewAnnotation>()
+  @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+  internal val idLookupMap = ConcurrentHashMap<View, String>()
+  private val hiddenViewMap = ConcurrentHashMap<View, Float>()
```

Comment:
- We have a lot of HashMaps here, can we refactor to a single? by introducing a model that encapsulates these states that are managed independently? 
- Why do we have concurrent? is this solution multi threaded?

Reply:
- user operates with actual views so we want O(1) to actually lookup internal structure for view annotation (we use `idLookupMap` for that). While core callbacks operate with ids and not views so we do use `annotationMap` with different key to also go with O(1) complexity. `hiddenViewMap` is a separate instance needed for non-trivial solution about view visibility, see https://github.com/mapbox/mapbox-maps-android/pull/847 for visuals.
 - we use separate thread but collection may be modified while we're iterating so to avoid concurrent modification exception we go with this impl